lib/logstash/util/unicode_trimmer.rb

```diff
@@ -0,0 +1,74 @@
+module UnicodeTrimmer
+  # The largest possible unicode chars are 4 bytes
+  # http://stackoverflow.com/questions/9533258/what-is-the-maximum-number-of-bytes-for-a-utf-8-encoded-character
+  # http://tools.ietf.org/html/rfc3629
+  MAX_CHAR_BYTES = 4
```

Comment:
Hah, I thought it was 6 also. The RFC indicates otherwise though. 5- and 6-byte sequences were removed (section 10 Security of the RFC) for security/buffer sizing problems, it seems.